### PR TITLE
[67] [Backend] As a user, I can save surveys to local

### PR DESCRIPTION
**Pull request by manh-t**
https://github.com/manh-t/flutter-survey/issues/67

## What happened 👀

The list of surveys must be fetched when opening the application and cached onto the device.

## Insight 📝

Use `Hive` as a local database to store surveys and user data.

## Proof Of Work 📹

CI should pass

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
LGTM 🤖

File: lib/api/local/database/user_box_helper.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  UserBoxHelperImpl(@Named('userBox') this._userBox);
+
+  User? get _user => _userBox.get(_userKey);
+
+  @override
+  Stream<User?> get user =>
+      _userBox.watch(key: _userKey).map((event) => _user).startWith(_user);
+
+  @override
+  void refreshUser(User user) {
```
Comment (sleepylee) [inline lib/api/local/database/user_box_helper.dart:33]: I feel like this is unnecessary as you only have 1 static `_userKey = 0` so if you `put` - it should override the old one no? 🤷  we don't have to `clear` actually

File: lib/main-staging.dart [hunk 2]
```diff
@@ -3,12 +3,14 @@ import 'dart:async';
...
 import 'package:flutter/services.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_survey/api/local/database/hive.dart';
 import 'package:flutter_survey/di/di.dart';
 import 'app.dart';
 import 'flavors.dart';
 
 Future<void> main() async {
   WidgetsFlutterBinding.ensureInitialized();
+  await initHive();
```
Comment (sleepylee) [inline lib/main-staging.dart:13]: can't we add these steps in the real `main.dart` file? 🤔

File: lib/api/local/database/user_box_helper.dart (hunk 1 above)
Comment (manh-t) [reply lib/api/local/database/user_box_helper.dart:33]: ah yesss, using `put` will override the old value 😓 my bad, will remove this redundant function 🙏

File: lib/main-staging.dart (hunk 2 above)
Comment (manh-t) [reply lib/main-staging.dart:13]: let me create a separate `main.dart` to handle duplication between `main-production.dart` and `main-staging.dart` 🙏

**Comment by manh-t** [comment]
@sleepylee thank you, I resolved them 🙏

**Review by sleepylee — APPROVED** [review]

**Pull request merged**